pkg/cache/v3/order_test.go

```diff
@@ -0,0 +1,49 @@
+package cache
+
+import (
+	"errors"
+	"sort"
+	"testing"
+
+	"github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+)
+
+func TestOrderKeys(t *testing.T) {
+	unorderedKeys := keys{
+		{
+			ID:      1,
+			TypeURL: resource.EndpointType,
+		},
+		{
+			ID:      2,
+			TypeURL: resource.ClusterType,
+		},
+		{
+			ID:      4,
+			TypeURL: resource.ListenerType,
+		},
+		{
+			ID:      3,
+			TypeURL: resource.RouteType,
+		},
+		{
+			ID:      5,
+			TypeURL: resource.ScopedRouteType,
+		},
+	}
+
+	orderedKeys := unorderedKeys
+	sort.Sort(orderedKeys)
+
+	if !sort.IsSorted(orderedKeys) {
```

Comment:
i dont think this really works to test it this way, its just checking itself with itself, right now I get this order when i print `orderedKeys` and the test passes:

```
{4 type.googleapis.com/envoy.config.listener.v3.Listener}
{3 type.googleapis.com/envoy.config.route.v3.RouteConfiguration}
{5 type.googleapis.com/envoy.config.route.v3.ScopedRouteConfiguration}
{2 type.googleapis.com/envoy.config.cluster.v3.Cluster}
{1 type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment}
```

Reply:
Yea so this test is pretty much pointless. It was really just to show that we can sort the keys in the order they're expected, LDS before RDS and SRDS, and CDS before EDS. I guess LDS before CDS doesn't matter though

Reply:
Also want to note here that when I started sending scoped routes before routes envoy began to complain. I wasn't sure if that was a bug so I moved it back and all seemed to be okay. I think the most important ordering here is that LDS and CDS send first so envoy can build its config tree properly.